pkg/cvo/cvo.go

```diff
@@ -281,32 +279,20 @@ func (optr *Operator) InitializeFromPayload(restConfig *rest.Config, burstRestCo
 // package for more details on the algorithm for verification. If the annotation is set, a verifier or error
 // is always returned.
 func loadConfigMapVerifierDataFromUpdate(update *payload.Update, clientBuilder sigstore.HTTPClient, configMapClient coreclientsetv1.ConfigMapsGetter) (verify.Interface, *verify.StorePersister, error) {
-	configMapGVK := corev1.SchemeGroupVersion.WithKind("ConfigMap")
-	for _, manifest := range update.Manifests {
-		if manifest.GVK != configMapGVK {
-			continue
-		}
-		if _, ok := manifest.Obj.GetAnnotations()[verify.ReleaseAnnotationConfigMapVerifier]; !ok {
-			continue
-		}
-		src := fmt.Sprintf("the config map %s/%s", manifest.Obj.GetNamespace(), manifest.Obj.GetName())
-		data, _, err := unstructured.NestedStringMap(manifest.Obj.Object, "data")
-		if err != nil {
-			return nil, nil, errors.Wrapf(err, "%s is not valid: %v", src, err)
-		}
-		verifier, err := verify.NewFromConfigMapData(src, data, clientBuilder)
-		if err != nil {
-			return nil, nil, err
-		}
-
-		// allow the verifier to consult the cluster for signature data, and also configure
-		// a process that writes signatures back to that store
-		signatureStore := configmap.NewStore(configMapClient, nil)
-		verifier.Store = &serial.Store{Stores: []store.Store{signatureStore, verifier.Store}}
-		persister := verify.NewSignatureStorePersister(signatureStore, verifier)
-		return verifier, persister, nil
+	verifier, err := verify.NewFromManifests(update.Manifests, clientBuilder)
+	if err != nil {
+		return nil, nil, err
 	}
```

Comment:
```suggestion
	verifier, err := verify.NewFromManifests(update.Manifests, clientBuilder)
	if err != nil || verifier == nil {
		return nil, nil, err
	}
```

Reply:
I don't think we need to be *that* defensive, but I'm ok adding language to `NewFromManifests`' docstring like "... returns a verifier or an error" to show that `err == nil` means `verifier` is populated.

Reply:
Since this is a code port to common I've tried to minimize changes. Per the original code and comment, it is not an error to return a nil verifier: "It returns an error if the data is not valid, or no verifier if no config map is found."

Reply:
Okay, lets leave it as is